test/pkg/environment/common/environment.go

```diff
@@ -129,6 +133,14 @@ func NewClient(ctx context.Context, config *rest.Config) client.Client {
 		})
 		return []string{t.Value}
 	}))
+	// drifted type
+	lo.Must0(cache.IndexField(ctx, &v1beta1.NodeClaim{}, "status.conditions[*].type", func(o client.Object) []string {
```

Comment:
Should we name this something that indicates that this is matching on the drift condition? It seems odd to me that we would name the indexer like this when it is only matching on the Drift type